src/app/3d/qgs3dmapcanvas.h

```diff
@@ -0,0 +1,44 @@
+#ifndef QGS3DMAPWIDGET_H
+#define QGS3DMAPWIDGET_H
+
+#include <QWidget>
+
+namespace Qt3DExtras
+{
+  class Qt3DWindow;
+}
+
+class Qgs3DMapSettings;
+class Qgs3DMapScene;
+
+
+class Qgs3DMapCanvas : public QWidget
+{
+    Q_OBJECT
+  public:
+    Qgs3DMapCanvas( QWidget *parent = nullptr );
+    ~Qgs3DMapCanvas();
+
+    //! Configure map scene being displayed. Takes ownership.
+    void setMap( Qgs3DMapSettings *map );
+
+    Qgs3DMapSettings *map() { return mMap; }
```

Comment:
Do you think that a const accessor could be useful in the future?

Reply:
Right now it is intentional that clients can get non-const instance of 3D map settings and modify it. This is slightly different from how it is done with 2D map canvas... Qgs3DMapSettings class emits signals when something has changed inside, so the 3D scene can react to that with least amount of disruption.